NYPLAudiobookToolkit/UI/AudiobookTableOfContentsTableViewController.swift

```diff
@@ -10,37 +10,94 @@ import UIKit
         self.tableOfContents = tableOfContents
+        self.delegate = delegate
+        self.activityIndicator = UIActivityIndicatorView(activityIndicatorStyle: .gray)
+        self.activityIndicator.hidesWhenStopped = true
         super.init(nibName: nil, bundle: nil)
-        let deleteItem = UIBarButtonItem(
-            barButtonSystemItem: .trash,
-            target: self,
-            action: #selector(AudiobookTableOfContentsTableViewController.deleteChapterRequested(_:))
-        )
+        //TODO localize when text is final (or replace with icon)
         let downloadAllItem = UIBarButtonItem(
-            barButtonSystemItem: .save,
-            target: self, action: #selector(AudiobookTableOfContentsTableViewController.downloadAllChaptersRequested(_:)))
-        self.navigationItem.rightBarButtonItems = [ downloadAllItem, deleteItem ]
+            title: "(Download All)",
+            style: .plain,
+            target: self,
+            action: #selector(AudiobookTableOfContentsTableViewController.downloadAllChaptersRequested(_:)))
+        let activityItem = UIBarButtonItem(
+            customView: self.activityIndicator)
+        self.navigationItem.rightBarButtonItems = [ downloadAllItem, activityItem ]
         self.tableOfContents.delegate = self
-        self.tableView.register(
-            UITableViewCell.self,
-            forCellReuseIdentifier: AudiobookTableOfContentsTableViewControllerCellIdentifier
-        )
         self.tableView.dataSource = self.tableOfContents
         self.tableView.delegate = self.tableOfContents
     }
     
     required public init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
+
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+        let container = UIView(frame: CGRect(x: 0, y: 0, width: self.tableView.frame.width, height: 80))
+        self.tableView.tableFooterView = container
+
+        let deleteButton = UIButton()
+        container.addSubview(deleteButton)
+
+        let title = NSLocalizedString("Clear Downloads", bundle: Bundle.audiobookToolkit()!, value: "Clear Downloads", comment: "Remove downloaded chapters from the device to save storage space")
+
+        deleteButton.autoCenterInSuperview()
+        deleteButton.setTitle(title, for: .normal)
+        deleteButton.setTitleColor(.red, for: .normal)
+        deleteButton.addTarget(self, action: #selector(deleteChapterRequested(_:)), for: .touchUpInside)
+    }
+
+    public override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        if let currentPlayingChapter = self.tableOfContents.player.currentChapterLocation {
+            let spine = self.tableOfContents.networkService.spine
+            for index in 0..<spine.count {
+                if currentPlayingChapter.inSameChapter(other: spine[index].chapter) {
```

Comment:
Here we only use data that come from the `AudiobookTableOfContents` object, so this functionality likely belongs in the `AudiobookTableOfContents` object.

something like this:

```swift 
// inside class AudiobookTableOfContents
func currentIndex() -> Int? {
  guard let currentPlayingChapter = self.tableOfContents.player.currentChapterLocation else { return nil }
  let spine = self.tableOfContents.networkService.spine
  for index in 0..<spine.count {
    if currentPlayingChapter.inSameChapter(other: spine[index].chapter) {
      return index
    }
  }
}
```

then the viewWillAppear just call this method and scrolls to it if we have a current index. The view does not need to know how this data is acquired.

Reply:
That's great, actually. I spent some time thinking about this and I'm not sure how I didn't come up with it. Thanks!
I think you can see my goal to get table view controller responsibilities like highlighting cells away from the data source. It took a little rejiggering.